index.html

```diff
@@ -37,304 +37,179 @@
-        {
-                canvas.toBlob(resolve); // implied image/png format
-        });
-};
+ImageData.create(blob).then(
+  imgData =&gt; doSomethingWith(imgData)
+);
 </pre>
       <p>
-        With the new method this can be simplified to:
+        This example demonstrates compressing an <code>ImageData</code> to a
+        <code>Blob</code>, and then displaying it in a
+        <code>HTMLImageElement</code>.
       </p>
       <pre class='example highlight'>
-function ImageDataToBlob(imageData)
-{
-        return imageData.toBlob();              // implied "image/png" format
-};
+var type = (Image.canEncodeType("image/jpg")) ? "image/jpg" : "image/png";
+imageData.toBlob(type).then((blob) =&gt; {
+  const img = new Image();
+  img.src = URL.createObjectURL(blob);
+  document.body.appendChild(img);
+});
 </pre>
     </section>
     <section>
       <h2>
-        Conversion to other types
-      </h2>
-      <p>
-        ImageData can already be converted to and from a canvas via
-        <code>getImageData</code> and <code>putImageData</code> (which probably
-        should remain synchronous to avoid racing with other drawing), and to
-        ImageBitmap via <code>createImageBitmap</code>. Conversion back to a
-        HTMLImageElement requires that there be a valid <code>src</code> for
-        the image, so this should be done via first converting to a Blob and
-        creating a URL to the blob.
-      </p>
-      <p>
-        Conversion from an ImageBitmap is omitted to avoid complications with
-        the fact implementations may store them on the GPU in premultiplied
-        form. Therefore converting an ImageBitmap to an ImageData could require
-        an expensive readback from the GPU, and make it difficult to avoid a
-        lossy pass through a premultiplied format. Instead whatever the
-        ImageBitmap was created from should be directly converted to an
-        ImageData without using ImageBitmap as an intermediate stage.
-      </p>
-      <p>
-        Note this proposal does not cover animated formats, in particular GIF,
-        APNG or MJPEG. It is suggested that ImageData remains as a
-        representation of a single static image, and another interface
-        providing access to an array of ImageData be specified for animated
-        formats with its own methods to convert to/from Blob. If an animated
-        format is specified for ImageData, it should operate on the first frame
-        only.
-      </p>
-    </section>
-    <section>
-      <h2>
-        Potential alternative solutions
+        Extensions to the <code>ImageData</code> class
       </h2>
+      <pre class="idl">
+typedef (HTMLImageElement or Blob) ImageSource;
+
+partial interface ImageData {
+  [NewObject] Promise&lt;Blob&gt; toBlob (optional DOMString type, optional EncoderOpts options);
+  [NewObject] static Promise&lt;ImageData&gt; create (ImageSource source);
+  static CanPlayTypeResult canDecodeType (DOMString mimeType);
```

Comment:
I don't think it should cover video or animated formats, at least in a v1 of the spec. ImageData represents a single still image so I think it should only cover still image formats. Video is best handled by HTMLVideoElement. ImageData could cover animated formats like MJPEG, APNG, GIF etc. but ImageData could only represent one frame (probably the first), which kind of defeats the point of using an animated format. In the "future work" section of the explainer document, I describe how I think it would be best to in future add some kind of AnimatedImageData interface which would deal with animated image formats (and possibly video? IDK) providing access to an array of ImageData for each frame, along with metadata.